internal/backend/s3/s3.go

```diff
@@ -215,6 +262,18 @@ func (be *Backend) Save(ctx context.Context, h restic.Handle, rd io.Reader) (err
 		return errors.New("key already exists")
 	}
 
+	// FIXME: This is a workaround once we move to minio-go 4.0.x this can be
+	// removed and size can be directly provided.
+	if f, ok := rd.(*os.File); ok {
+		debug.Log("reader is %#T, using nopCloserFile{}", rd)
+		rd = nopCloserFile{f}
+	} else if l, ok := rd.(lenner); ok {
+		debug.Log("reader is %#T, using nopCloserLenner{}", rd)
+		rd = nopCloserLenner{l}
+	} else {
+		debug.Log("reader is %#T, no specific workaround enabled", rd)
+	}
+
 	// prevent the HTTP client from closing a file
 	rd = ioutil.NopCloser(rd)
```

Comment:
This needs to be removed in order for the workaround above to be active :)

Reply:
oops yes..